kaitai/kaitaistream.cpp

```diff
@@ -344,6 +344,12 @@ uint64_t kaitai::kstream::get_mask_ones(int n) {
 std::string kaitai::kstream::read_bytes(std::streamsize len) {
     std::vector<char> result(len);
 
+    // NOTE: streamsize type is signed, negative values are only *supposed* to not be used.
+    // http://en.cppreference.com/w/cpp/io/streamsize
+    if (len < 0) {
+        throw std::runtime_error("read_bytes: requested a negative amount");
+    }
+
     if (len > 0 ) {
```

Comment:
This check should be actually `len != 0`. Thus we'll do a `read()` with negative amount of bytes, and that will result in clear native C++ exception.

Reply:
Um, actually, I'm kind of puzzled at why this check is even needed in the first place. At least where I've checked, it looks like it's perfectly ok to call `read()` requesting 0 bytes to read. @crackedmind, could you elaborate?